framework/src/modules/chain/components/storage/entities/transaction.js

```diff
@@ -0,0 +1,288 @@
+ * @property {Object} asset
+ * @property {Object} asset.delegate
+ * @property {string} asset.delegate.username
+ */
+
+/**
+ * Vote Transaction
+ * @typedef {BasicTransaction} VoteTransaction
+ * @property {Object} asset
+ * @property {Array.<string>} asset.votes
+ */
+
+/**
+ * Multisig Registration Transaction
+ * @typedef {BasicTransaction} MultisigRegistrationTransaction
+ * @property {Object} asset
+ * @property {Object} asset.multisignature
+ * @property {Integer} asset.multisignature.min
+ * @property {Integer} asset.multisignature.lifetime
+ * @property {Array.<string>} asset.multisignature.keysgroup
+ */
+
+/**
+ * Dapp Registration Transaction
+ * @typedef {BasicTransaction} DappRegistrationTransaction
+ * @property {Object} asset
+ * @property {Object} asset.dapp
+ * @property {Integer} asset.dapp.type
+ * @property {string} asset.dapp.name
+ * @property {string} asset.dapp.description
+ * @property {string} asset.dapp.tags
+ * @property {string} asset.dapp.link
+ * @property {string} asset.dapp.icon
+ * @property {Integer} asset.dapp.category
+ */
+
+/**
+ * InTransfer Transaction
+ * @typedef {BasicTransaction} InTransferTransaction
+ * @property {Object} asset
+ * @property {Object} asset.inTransfer
+ * @property {string} asset.inTransfer.dappId
+ */
+
+/**
+ * OutTransfer Transaction
+ * @typedef {BasicTransaction} OutTransferTransaction
+ * @property {Object} asset
+ * @property {Object} asset.outTransfer
+ * @property {string} asset.outTransfer.dappId
+ * @property {string} asset.outTransfer.transactionId
+ */
+
+/**
+ * Transaction
+ * @typedef {(TransferTransaction|SecondPassphraseTransaction|DelegateTransaction|VoteTransaction|MultisigRegistrationTransaction|DappRegistrationTransaction|InTransferTransaction|OutTransferTransaction)} Transaction
+ */
+
+/**
+ * Transaction Filters
+ * @typedef {Object} filters.Transaction
+ */
+
+const sqlFiles = {
+	create: 'transactions/create.sql',
+};
+
+const trsCreateFields = [
+	'id',
+	'blockId',
+	'type',
+	'timestamp',
+	'senderPublicKey',
+	'requesterPublicKey',
+	'senderId',
+	'recipientId',
+	'amount',
+	'fee',
+	'signature',
+	'signSignature',
+	'signatures',
+	'asset',
+	'transferData',
+];
+
+class Transaction extends TransactionEntity {
+	/**
+	 * Constructor
+	 * @param {BaseAdapter} adapter - Adapter to retrive the data from
+	 * @param {filters.Transaction} defaultFilters - Set of default filters applied on every query
+	 */
+	constructor(adapter, defaultFilters = {}) {
+		super(adapter, defaultFilters);
+
+		this.sqlDirectory = path.join(path.dirname(__filename), '../sql');
+
+		this.SQLs = this.loadSQLFiles('transaction', sqlFiles, this.sqlDirectory);
+	}
+
+	/**
+	 * Create transactions object
+	 *
+	 * @param {Transaction|Array.<Transaction>} data
+	 * @param {Object} [_options]
+	 * @param {Object} [tx] - Transaction object
+	 * @return {*}
+	 */
+	create(data, _options, tx) {
+		const transactions = Transaction._sanitizeCreateData(data);
```

Comment:
`Transaction._sanitizeCreateData` implementation should be moved from `components/storage` to this file